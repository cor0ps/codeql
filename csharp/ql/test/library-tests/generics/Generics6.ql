/**
 * @name Test for generics
 */

import csharp

from ConstructedClass at, UnboundGenericClass b, ConstructedClass bt, Field f
where
  at.hasName("A<T>") and
  b.hasName("B<>") and
  bt.hasName("B<X>") and
  at.getTypeArgument(0).hasName("T") and
  at.getTypeArgument(0) instanceof TypeParameter and
  at.getTypeArgument(0) = b.getTypeParameter(0) and
  bt.getUnboundGeneric() = b and
  f.getDeclaringType() = b and
  f.getType() = at
select b, bt, f, at
