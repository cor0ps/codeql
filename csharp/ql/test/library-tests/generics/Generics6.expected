| generics.cs:22:18:22:21 | B<> | generics.cs:22:18:22:21 | B<X> | generics.cs:25:23:25:24 | at | generics.cs:13:18:13:21 | A<T> |
